src/test/java/com/jcabi/ssh/SSHTest.java

```diff
@@ -102,6 +111,45 @@ public void executeCommandOnServer() throws Exception {
     }
 
     /**
+     * SSH can execute command on ssh server with private key.
+     * @throws Exception In case of error.
+     */
+    @Test
+    public void executeCommandOnServerWithPrivateKey() throws Exception {
+        final int port = SSHTest.port();
+        final SshServer sshd = new MockSshServerBuilder(port)
+            .usePublicKeyAuthentication().build();
+        sshd.setCommandFactory(new MkCommandCreator());
+        sshd.start();
+        final String cmd = SOME_TEST_COMMAND;
+        final ByteArrayOutputStream output = new ByteArrayOutputStream();
+        final SSHOptions sshOptions = new SSHOptions.Builder()
+            .setAddr(InetAddress.getLocalHost().getCanonicalHostName())
+            .setPort(port)
+            .setLogin(TEST)
+            .setPrivatekey(
+                IOUtils.toString(
+                    this.getClass()
+                        .getResourceAsStream("encrypted_private.key")
+                )
+            )
+            .setPassphrase("test-passphrase")
+            .build();
+        final Shell shell = new Shell.Verbose(
+            new SSH(sshOptions)
+        );
+        final int exit = shell.exec(
+            cmd,
+            new NullInputStream(0L),
+            output,
+            Logger.stream(Level.WARNING, true)
+        );
+        sshd.stop();
+        MatcherAssert.assertThat(exit, Matchers.equalTo(0));
+        MatcherAssert.assertThat(output.toString(), Matchers.equalTo(cmd));
```

Comment:
@LinkedList there is a short version of `Matchers.equalTo(cmd)` -> `Matchers.is(cmd)`